### PR TITLE
Update latest-posts.php

### PR DESCRIPTION
**Pull request by mujuonly**
Removed unwanted space in the string

Trac ticket: https://core.trac.wordpress.org/ticket/59409

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/blocks/latest-posts.php [hunk 1]
```diff
@@ -51,7 +51,7 @@ function render_block_core_latest_posts( $attributes ) {
 	$filter_latest_posts_excerpt_more = static function( $more ) use ( $attributes ) {
 		$use_excerpt = 'excerpt' === $attributes['displayPostContentRadio'];
 		/* translators: %1$s is a URL to a post, excerpt truncation character, default … */
-		return $use_excerpt ? sprintf( __( ' … <a href="%1$s" rel="noopener noreferrer">Read more</a>' ), esc_url( get_permalink() ) ) : $more;
+		return $use_excerpt ? sprintf( __( '… <a href="%1$s" rel="noopener noreferrer">Read more</a>' ), esc_url( get_permalink() ) ) : $more;
```
Comment (mukeshpanchal27) [inline src/wp-includes/blocks/latest-posts.php]: @mujuonly Default WP also add space before after the excerpt content 🤔

Comment (mujuonly) [reply src/wp-includes/blocks/latest-posts.php]: Screenshot 2023-09-20 at 3 05 41 PM
:man-shrugging:

**Comment by mukeshpanchal27** [comment]
@SergeyBiryukov @costdev Could you please take a look.

File: src/wp-includes/blocks/latest-posts.php (hunk 1 above)
Comment (costdev) [reply src/wp-includes/blocks/latest-posts.php]: I'm tempted to say that this should actually be:

```php
if ( $use_excerpt ) {
	return sprintf(
		' [&hellip;] <a href="%1$s" rel="noopener noreferrer">%2$s</a>',
		esc_url( get_permalink() ),
		__( 'Read more' )
	);
} else {
	return $more;
}
```
- Translators should not have to deal with markup in the string to be translated.
- This retains the space before and after the `...`, and adds `[]` for consistency with [the default in Core](https://github.com/costdev/wordpress-develop/blob/2a8f8eca77b7c54ce2a3f04a37c064ae7f8a6552/src/wp-includes/formatting.php#L4017).
- The `'Read more'` string [is already translated](https://github.com/costdev/wordpress-develop/blob/2a8f8eca77b7c54ce2a3f04a37c064ae7f8a6552/src/wp-includes/blocks/read-more.php#L37), so we should leverage this.

Comment (ramonjd) [reply src/wp-includes/blocks/latest-posts.php]: Thanks for working on this! 

I'm looking at updating the string and a couple of semi-related issues in the latest posts block over in Gutenberg in https://github.com/WordPress/gutenberg/pull/55029/files#diff-55d79982c1f505132fcff0c29e1e84b411c56fbbf5dc1c55220a530125ebf8bdR489

File: src/wp-includes/blocks/latest-posts.php [hunk 2]
```diff
@@ -50,8 +50,16 @@ function render_block_core_latest_posts( $attributes ) {
 
 	$filter_latest_posts_excerpt_more = static function( $more ) use ( $attributes ) {
 		$use_excerpt = 'excerpt' === $attributes['displayPostContentRadio'];
-		/* translators: %1$s is a URL to a post, excerpt truncation character, default … */
-		return $use_excerpt ? sprintf( __( ' … <a href="%1$s" rel="noopener noreferrer">Read more</a>' ), esc_url( get_permalink() ) ) : $more;
+		if ( $use_excerpt ) {
+			/* translators: %1$s is a URL to a post, excerpt truncation character, default … */
```
Comment (costdev) [inline src/wp-includes/blocks/latest-posts.php:54]:
```suggestion
```
The translators comment is no longer needed as only `'Read more'` is translated here, and it includes no tokens to be explained. 🙂

**Comment by swissspidy** [comment]
This was fixed in Gutenberg https://github.com/WordPress/gutenberg/commit/8ef6e404f1de2ca22e3921c8e454e4335c843143

**Pull request closed without merging**